File: 1537-maximum-score-after-splitting-a-string/1537-maximum-score-after-splitting-a-string.cs
public class Solution {
    public int MaxScore(string s) {
        
        int sum = 0;
        foreach(var c in s) sum += int.Parse(c.ToString());

        int cSum = 0, res = 0;
        for (int i = 0; i < s.Length - 1; i++) {
            cSum += int.Parse(s[i].ToString());
            int numZeroes = i - cSum + 1;
            int numOnes = sum - cSum;
            res = Math.Max(res, numOnes + numZeroes);
        }

        return res;
    }
}

File: 1537-maximum-score-after-splitting-a-string/1537-maximum-score-after-splitting-a-string.py
class Solution:
    def maxScore(self, s: str) -> int:

        total_ones = 0
        for c in s:
            if c == "1":
                total_ones += 1
        res = 0
        
        num_of_zeroes = 0
        num_of_ones = 0

        for i in range(len(s) - 1):
            c = s[i]
            if c == "0":
                num_of_zeroes += 1
            else:
                num_of_ones += 1
            print(num_of_ones)
            print(num_of_zeroes)
            print("------")
            res = max(res, total_ones - num_of_ones + num_of_zeroes)
        
        return res
            
        